packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -318,7 +334,21 @@ class AnimationController extends Animation<double>
   /// The most recently returned [TickerFuture], if any, is marked as having been
   /// canceled, meaning the future never completes and its [TickerFuture.orCancel]
   /// derivative future completes with a [TickerCanceled] error.
+  ///
+  /// During the animation, [status] is reported as [AnimationStatus.forward]
+  /// regardless of whether `target` > [value] or not. At the end of the
+  /// animation, when `target` is reached, [status] is reported as
+  /// [AnimationStatus.completed].
+  ///
+  /// You should not call [forward] or [reverse] on the same instance of an
+  /// [AnimationController] on which you also call [animateTo]. If you do mix
+  /// those calls, [status] might not have the expected value.
```

Comment:
Except that now that you've made this change, why won't it have the right value? You've removed the confusion, no?

This seems like a great change, if none of the tests fail.